### DISCUSSION TITLE
Is it possible to allow images from a specific directory in my domain only?

### DISCUSSION THREAD

**Discussion by supercain** [apostrophecms/sanitize-html#536] [Q&A]
Hi,

I want to allow only images that are located here http://example.com/images/ so I did this:

allowedSchemes: ['images/']

but it doesnt seem to work since it allows images from any directory. Is there a way to do it?

Thank you.

**Comment by boutell** [comment]
This option is for URL schemes (protocols), not folders. This doesn't mean there's no solution to your problem though. I don't have an answer offhand, but I suggest that you look at the various other filtering and transformation options in the documentation for ideas.

**Reply by supercain** [reply]
Hi, is there a full list of all the options available? Seems like there is only a few on the documentation page.

**Comment by boutell** [comment]
Check out the "transformations" section, which covers what you're looking to do.

**Reply by supercain** [reply]
Hi again,

I tried this because it sounded like what I am looking for:

```
allowedTags: [ 'span', 'img' ],
       allowedAttributes: {
        'span': [ 'style' ], 
        'img': [
          {
            name: 'src',
            values: [/^\.\.\/public\/style_emoticons\/emoticons\/+.*$/]
          }
        ]
       }
```

which should allow all images under ../public/style_emoticons/emoticons/ but it wont parse it.

**Reply by supercain** [reply]
It doesnt make any sense, I tested the regex and its correct.

**Comment by boutell** [comment]
My first question is how you can have unescaped slashes in the middle of a
regular expression. Sure enough you said it wouldn't parse it — I think you
mean javascript wouldn't parse it.

Second, it is unlikely that your URLs have "../public" in them, although it
is not impossible. That sounds more like a filesystem relative path to
where your files are on the hard drive, which is not the same thing. You're
validating what is in the src attribute, not a filesystem path.


On Tue, Feb 8, 2022 at 5:22 PM supercain ***@***.***> wrote:

> It doesnt make any sense, I tested the regex and its correct.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apostrophecms/sanitize-html/discussions/536#discussioncomment-2136895>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAH27LQEVQHTRLIZMS2YPTU2GJQRANCNFSM5NZLLV4Q>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***
> com>
>


-- 

THOMAS BOUTELL | CHIEF TECHNOLOGY OFFICER
APOSTROPHECMS | apostrophecms.com | he/him/his

**Reply by supercain** [reply]
The slashes are escaped, but they wouldnt get parsed as I posted the code here the first time, they look alright now. Also, I know its unusual but thats how my directory looks. I already managed to solve this issue anyway by using tranform as you suggested. Thanks.
